flocker/node/_docker.py

```diff
@@ -561,7 +563,18 @@ def _list():
                 # Retrieve environment variables for this container,
                 # disregarding any environment variables that are part
                 # of the image, rather than supplied in the configuration.
-                image_data = self._client.inspect_image(image)
+                try:
+                    image_data = self._client.inspect_image(image)
+                except APIError as e:
+                    if e.response.status_code == NOT_FOUND:
+                        # Image has been deleted, so just fill in some
+                        # stub data so we can return *something*. This
+                        # should happen only for stopped containers so
+                        # some inaccuracy is acceptable.
+                        image_data = {u"Config": {u"Env": []}}
+                    else:
```

Comment:
Also see https://github.com/ClusterHQ/flocker/pull/1340 - as in that case, as a temporary fix for a problem, I think this is fine, but also strongly suggests we need to re-think our approach in the `list` method and how we're representing default and missing information about containers and states from Docker.